source/MapPanel.cpp

```diff
@@ -69,13 +69,27 @@ namespace {
 	const unsigned MAX_MISSION_POINTERS_DRAWN = 12;
 	const double MISSION_POINTERS_ANGLE_DELTA = 30.;
 
-	// Struct to track per system how many pointers are drawn and still
+	// Class to track per system how many pointers are drawn and still
 	// need to be drawn.
-	struct PointerDrawCount {
+	class PointerDrawCount {
+	public:
+		// Calculate and check the most number of pointer positions that should be available for active missions.
+		// This can be up to half the maximum number of pointers that can be drawn.
+		void Reserve()
+		{
+			maximumActive = max(MAX_MISSION_POINTERS_DRAWN / 2, MAX_MISSION_POINTERS_DRAWN - (available + unavailable));
+		}
+
+		unsigned MaximumActive() const { return maximumActive; }
```

Comment:
There's the style rule that says no function bodies inside the class body, but there are other cases of this in the current code, so we can leave it for another PR.

Reply:
It still has to be on a separate line in this case, right?
```suggestion
		void Reserve()
		{
			maximumActive = max(MAX_MISSION_POINTERS_DRAWN / 2, MAX_MISSION_POINTERS_DRAWN - (available + unavailable));
		}

		unsigned MaximumActive() const
		{
			return maximumActive;
		}
```

Also, is there any reason to make this into a class if we just put public fields in it?

Reply:
I've moved the definitions out of the class.

> Also, is there any reason to make this into a class if we just put public fields in it?

It has one private field and it has a method handling it. ~~We also just shouldn't use the "struct" keyword per the style guide.~~

Reply:
Those review-comments appear to be handled.